### DISCUSSION TITLE
Array to string conversion on radio fields

### DISCUSSION THREAD

**Discussion by corneliusio** [solspace/craft-freeform#617] [Questions & Answers] [answered]
Have a site where any form submission with a radio field throws an error attempting to coerce the value to a string.

**Steps to reproduce**
1. Create form with radio field
2. Submit form on front-end

**Craft & Plugin Info (please complete the following information):**
Tested before and after updating Craft and Freeform. Same results in each case.

- Craft Version: 3.7.48 and 3.7.67
- Freeform Version: 3.12.11 and 3.13.24
- Freeform Edition: Pro
- Fresh Install or Upgrade: Upgrade

**Additional context**
```
yii\base\ErrorException: Array to string conversion in /Users/cornelius/Git/@mostlyserious/smsi/vendor/solspace/craft-freeform/packages/plugin/src/Library/Composer/Components/Fields/Traits/SingleValueTrait.php:29
Stack trace:
#0 /Users/cornelius/Git/@mostlyserious/smsi/vendor/craftcms/cms/src/web/ErrorHandler.php(84): yii\base\ErrorHandler->handleError(8, 'Array to string...', '/Users/corneliu...', 29)
#1 /Users/cornelius/Git/@mostlyserious/smsi/vendor/solspace/craft-freeform/packages/plugin/src/Library/Composer/Components/Fields/Traits/SingleValueTrait.php(29): craft\web\ErrorHandler->handleError(8, 'Array to string...', '/Users/corneliu...', 29, Array)
#2 /Users/cornelius/Git/@mostlyserious/smsi/vendor/solspace/craft-freeform/packages/plugin/src/Library/Composer/Components/Fields/Traits/SingleValueTrait.php(50): Solspace\Freeform\Fields\RadioGroupField->getValue()
#3 /Users/cornelius/Git/@mostlyserious/smsi/vendor/solspace/craft-freeform/packages/plugin/src/Bundles/Form/Context/Request/PostContext.php(46): Solspace\Freeform\Fields\RadioGroupField->setValue(Array)
#4 [internal function]: Solspace\Freeform\Bundles\Form\Context\Request\PostContext->handleRequest(Object(Solspace\Freeform\Events\Fields\TransformValueEvent))
#5 /Users/cornelius/Git/@mostlyserious/smsi/vendor/yiisoft/yii2/base/Event.php(312): call_user_func(Array, Object(Solspace\Freeform\Events\Forms\HandleRequestEvent))
#6 /Users/cornelius/Git/@mostlyserious/smsi/vendor/solspace/craft-freeform/packages/plugin/src/Library/Composer/Components/Form.php(745): yii\base\Event::trigger('Solspace\\Freefo...', 'before-handle-r...', Object(Solspace\Freeform\Events\Forms\HandleRequestEvent))
#7 /Users/cornelius/Git/@mostlyserious/smsi/vendor/solspace/craft-freeform/packages/plugin/src/Controllers/SubmitController.php(56): Solspace\Freeform\Library\Composer\Components\Form->handleRequest(Object(craft\web\Request))
#8 [internal function]: Solspace\Freeform\Controllers\SubmitController->actionIndex()
#9 /Users/cornelius/Git/@mostlyserious/smsi/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#10 /Users/cornelius/Git/@mostlyserious/smsi/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#11 /Users/cornelius/Git/@mostlyserious/smsi/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('', Array)
#12 /Users/cornelius/Git/@mostlyserious/smsi/vendor/craftcms/cms/src/web/Application.php(293): yii\base\Module->runAction('freeform/submit', Array)
#13 /Users/cornelius/Git/@mostlyserious/smsi/vendor/craftcms/cms/src/web/Application.php(602): craft\web\Application->runAction('freeform/submit', Array)
#14 /Users/cornelius/Git/@mostlyserious/smsi/vendor/craftcms/cms/src/web/Application.php(272): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#15 /Users/cornelius/Git/@mostlyserious/smsi/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
#16 /Users/cornelius/Git/@mostlyserious/smsi/web/index.php(21): yii\base\Application->run()
#17 /Users/cornelius/.composer/vendor/laravel/valet/server.php(235): require('/Users/corneliu...')
#18 {main}
```

**Comment by kjmartens** [comment]
Hi @corneliusio,

Can you share what your template code looks like? Any chance you are including array brackets (`[]`) in the radio field markup? 🙂

**Comment by corneliusio** [comment]
For sure, there's actually no custom templating going on, and the handle/name for the field has no special characters of any kind. Template code is just `{{ block.freeform.one.form.render }}`.

**Comment by kjmartens** [comment]
Are you using a default Freeform formatting template? Or a custom one? If custom, can you share that here as well? 🙂

**Comment by corneliusio** [comment]
🤦🏼 So sorry. I did not originally develop this site and just totally missed that it was using a custom template. The hard-coded array brackets were exactly the problem. I never saw anything related to the template in the stack trace and just didn't dig deep enough. 

Appreciate your help getting to the bottom of that. It appears as though I would still need to include the array brackets on `checkbox` fields and presumably also multi `select` fields and possibly `file` as well?

(Accidently typed this up this morning and apparently never submitted it. Sorry for the delay.)

**Comment by kjmartens** [comment, marked as answer]
If you're using automated things like `render`, etc, Freeform will take care of the `[]` automatically for you. If you're hardcoding the formatting template in some way, you'll need to make sure that the only field names that get `[]` are **Checkbox Group**, **Multi-Select** and **File Upload** fields (if you set it to allow more than 1 file). 🙂

**Reply by corneliusio** [reply]
Fantastic, thanks for your help getting to the heart of the issue we were seeing.
